contracts/ERC5725.sol

```diff
@@ -27,16 +30,39 @@ abstract contract ERC5725 is IERC5725, ERC721 {
      * @dev See {IERC5725}.
      */
     function claim(uint256 tokenId) external override(IERC5725) validToken(tokenId) {
-        require(ownerOf(tokenId) == msg.sender, "Not owner of NFT");
+        require(
+            ownerOf(tokenId) == msg.sender ||
+                (ownerOf(tokenId) != address(0) && allowance(ownerOf(tokenId), msg.sender) > 0),
+            "ERC5725: not owner of NFT or no permission to spend"
+        );
+
         uint256 amountClaimed = claimablePayout(tokenId);
         require(amountClaimed > 0, "ERC5725: No pending payout");
 
+        if (ownerOf(tokenId) != msg.sender) {
```

Comment:
SELF REVIEW : 

I feel like this could be expressed better, but not sure how.

At this point, we're either the owner, or a spender with balance > 0.
So if we're not the owner, then we must be the spender - I didn't think it was worth calculating if we're the spender again, because we've covered it in the above line.

Reply:
Really picking my brain on this one. I will have to get a rain check for now

Reply:
This can be removed with the suggested refactor I provided.